File: mahaam-api-py/infra/configs.py
from pydantic import BaseModel
import json
import logging


class Config(BaseModel):
    apiName: str
    apiVersion: str
    envName: str
    dbUrl: str
    logFile: str
    logFileSizeLimit: int
    logFileCountLimit: int
    logFileOutputTemplate: str
    logFileRollingInterval: str
    httpPort: int
    tokenSecretKey: str
    emailAccountSid: str
    emailVerificationServiceSid: str
    emailAuthToken: str
    testEmails: list[str]
    testSID: str
    testOTP: str
    logReqEnabled: bool


data: Config = None


def init(path: str):
    global data
    try:
        with open(path) as f:
            config_data = json.load(f)
            data = Config(**config_data)
    except Exception as e:
        logging.fatal("Failed to load config: %s", e)
        raise

File: mahaam-api-go-way/internal/pkg/configs/configs.go
package configs

import (
	"encoding/json"
	"log"
	"os"
)

var (
	ApiName                     string
	ApiVersion                  string
	EnvName                     string
	DBUrl                       string
	LogFile                     string
	LogFileSizeLimit            int
	LogFileCountLimit           int
	LogFileOutputTemplate       string
	LogFileRollingInterval      string
	HTTPPort                    int
	TokenSecretKey              string
	EmailAccountSID             string
	EmailVerificationServiceSID string
	EmailAuthToken              string
	TestEmails                  []string
	TestSID                     string
	TestOTP                     string
	LogReqEnabled               bool
)

func InitConfigs() {
	file, _ := os.Open("config.json")
	decoder := json.NewDecoder(file)
	conf := config{}
	err := decoder.Decode(&conf)
	if err != nil {
		log.Fatal("Error loading config file" + err.Error())
	}
	ApiName = conf.ApiName
	ApiVersion = conf.ApiVersion
	EnvName = conf.EnvName
	DBUrl = conf.DBUrl
	LogFile = conf.LogFile
	LogFileSizeLimit = conf.LogFileSizeLimit
	LogFileCountLimit = conf.LogFileCountLimit
	LogFileOutputTemplate = conf.LogFileOutputTemplate
	LogFileRollingInterval = conf.LogFileRollingInterval
	HTTPPort = conf.HTTPPort
	TokenSecretKey = conf.TokenSecretKey
	EmailAccountSID = conf.EmailAccountSID
	EmailVerificationServiceSID = conf.EmailVerificationServiceSID
	EmailAuthToken = conf.EmailAuthToken
	TestEmails = conf.TestEmails
	TestSID = conf.TestSID
	TestOTP = conf.TestOTP
	LogReqEnabled = conf.LogReqEnabled
}

type config struct {
	ApiName                     string
	ApiVersion                  string
	EnvName                     string
	DBUrl                       string
	LogFile                     string
	LogFileSizeLimit            int
	LogFileCountLimit           int
	LogFileOutputTemplate       string
	LogFileRollingInterval      string
	HTTPPort                    int
	TokenSecretKey              string
	EmailAccountSID             string
	EmailVerificationServiceSID string
	EmailAuthToken              string
	TestEmails                  []string
	TestSID                     string
	TestOTP                     string
	LogReqEnabled               bool
}
